tests/fib_macro.pil

```diff
@@ -0,0 +1,33 @@
+constant %N = 16;
+
+namespace Fibonacci(%N);
+    constant %last_row = %N - 1;
+    // TODO what do we do if macros generate identities for numbers?
+    // -> We verify them, and if they hold, we ignore them.
+    macro bool(X) { X * (1 - X) = 0; };
+    macro is_nonzero(X) { X / X }; // 0 / 0 == 0 makes this work...
+    macro is_zero(X) { 1 - is_nonzero(X) };
+    macro is_equal(A, B) { is_zero(A - B) };
+    macro is_one(X) { is_equal(X, 1) };
+    macro ite(C, A, B) { is_nonzero(C) * A + is_zero(C) * B};
```

Comment:
Shouldn't the macros above have `= 0` for all expression, like `bool` does?

Reply:
It depends if they evaluate to an expression or a constraint. If you use `=`, the result is a polynomial identity constraint and you cannot continue building more expressions with it. That's why `is_equal` evaluates to `1` if the argumenst are equal and `0` otherwise.

Having said that, I'm not too happy with that either. We can iterate on it later.

Reply:
Yea I think this effectively creates a difference in things that are macros and others that are functions